contributed_definitions/nyaml/NXsubstance.yaml

```diff
@@ -0,0 +1,41 @@
+category: base
+doc: |
+  A form of matter with a constant, definite chemical composition.

```

Comment:
Alloys are really a corner case you are right provided that one accepts or states then I am saying that specific alloy is an NXsubstance instance because I do not care in my application for the internal structure, the problem is an alloy in most cases as atomically individually different definite chemical composition (if at all) and thus in this case it must not be modelled eventually as an Nxsubstance but a composite of substances this offers such a good combination for users with different levels of expressiveness